brickflow_plugins/databricks/workflow_dependency_sensor.py

```diff
@@ -171,3 +199,100 @@ def execute(self):
 
             self.log.info(f"sleeping for: {self.poke_interval}")
             time.sleep(self.poke_interval)
+
+
+class WorkflowTaskDependencySensor(WorkflowDependencySensor):
+    """
+    This is used to have dependencies on the specific task within a databricks workflow
+
+    Example Usage in your brickflow task:
+        service_principle_pat = ctx.dbutils.secrets.get("scope", "service_principle_id")
+        WorkflowDependencySensor(
+            databricks_host=https://your_workspace_url.cloud.databricks.com,
+            databricks_token=service_principle_pat,
+            dependency_job_id=job_id,
+            dependency_task_name="foo",
+            poke_interval=20,
+            timeout=60,
+            delta=timedelta(days=1)
+        )
+        In the above snippet Databricks secrets are used as a secure service to store the databricks token.
+        If you get your token from another secret management service, like AWS Secrets Manager, GCP Secret Manager
+        or Azure Key Vault, just pass it in the databricks_token argument.
+    """
+
+    def __init__(
+        self,
+        databricks_host: str,
+        databricks_token: Union[str, SecretStr],
+        dependency_job_name: str,
+        dependency_task_name: str,
+        delta: timedelta,
+        timeout_seconds: int,
+        poke_interval_seconds: int = 60,
+    ):
+        super().__init__(
+            databricks_host=databricks_host,
+            databricks_token=databricks_token,
+            dependency_job_name=dependency_job_name,
+            delta=delta,
+            timeout_seconds=timeout_seconds,
+            poke_interval_seconds=poke_interval_seconds,
+        )
+
+        self.dependency_task_name = dependency_task_name
+
+    def execute(self):
+        self.dependency_job_id = self._get_job_id
+
+        session = self.get_http_session()
+        url = f"{self.databricks_host.rstrip('/')}/api/2.1/jobs/runs/list"
```

Comment:
Can we make use of WorkspaceClient() so that if the api versions change then we don't need to modify the code?

```
def get_job_id_v2(job_name: str) -> int:
    """
    This function is used to retrieve the job ID of a specific job from Databricks using the WorkspaceClient
    :param job_name: The name of the job to find
    :return: The job id if found, otherwise raises an Exception
    """
    try:
        workspace_obj = WorkspaceClient(host="", token="")
        jobs_list = workspace_obj.jobs.list()

        for job in jobs_list:
            if job.settings.name == job_name:
                print(job.job_id)
                return job.job_id
        else:
            raise Exception(f"No job found with name {job_name}")
    except Exception as e:
        print(f"An error occurred: {e}")
```

Reply:
@shwen-gine @asingamaneni updated the code to use SDK

Reply:
Thank you!